public/js/search.js

```diff
@@ -0,0 +1,24 @@
+$(document).ready(() => {
+    // remove search in nav bar on the search page
+    if (window.location.href.includes('search')) {
+        $('#nav-search').remove();
+    }
+
+    if (window.location.href.includes('users/search')) {
+        $('#search_type').val('Users');
+        $('#search-box').attr('action', '/users/search');
+    }
+
+    $('#search_type').change((e) => {
```

Comment:
Instead of dynamically changing endpoint, we can just pass resource as parameter

/search?query='Ankit'&resource=Project

Reply:
Currently I have moved the search routes of users and projects to their respective controller(and have their own routes). They also have their own specific query implementations. If we move to the route you suggested, how are we planning to detect which query to call in the backend? Are we planning to have a generic query implementation for both users and projects?

Reply:
I think we can have generic implementation

Reply:
Currently, Users and Projects are the only two models that we have to search, will a generic query implementation be helpful? 🤔

Reply:
you can use dependency injection to specify the search query based on resource

Reply:
So should we have a separate search controller with this route and decide the query to use on the basis of the resource key passed?

Reply:
Moved search logic to the search controller. 😄